gson/src/test/java/com/google/gson/JsonArrayTest.java

```diff
@@ -99,4 +102,65 @@ public void testDeepCopy() {
     assertEquals(1, original.get(0).getAsJsonArray().size());
     assertEquals(0, copy.get(0).getAsJsonArray().size());
   }
+
+  public void testFailedGetArrayValues() {
+    String arrayStr = "[" + "true," + "false," + "\"true\"," + "\"false\"," + "\"hello\"," + "23.45e-4," + "\"23.45\"," + "42," + "\"43\"," + "[" + "\"world\"" + "]," + "{" + "\"key1\":\"value1\"," + "\"key2\":\"value2\"," + "\"key3\":\"value3\"," + "\"key4\":\"value4\"" + "}," + "0," + "\"-1\"" + "]";
+    JsonArray jsonArray = (JsonArray) JsonParser.parseString(arrayStr);
+    try {
+      jsonArray.get(10).getAsBoolean();
+      assertTrue("expected getBoolean to fail", false);
+    } catch (UnsupportedOperationException e) {
+      assertEquals("Expected an exception message",
+              "JsonObject",e.getMessage());
```

Comment:
Really minor, but for consistent code style there should probably be a space:
```suggestion
              "JsonObject", e.getMessage());
```
(applies to other tests as well)

Reply:
Resolved the formatting issues.